packages/core/src/lib/markdown-it/highlight/HighlightRule.js

```diff
@@ -0,0 +1,73 @@
+const { HighlightRuleComponent } = require('./HighlightRuleComponent.js');
+
+class HighlightRule {
+  constructor(rule) {
+    /**
+     * @type Array<HighlightRuleComponent>
+     */
+    this.rule = rule;
+  }
+  
+  isLineRange() {
+    return this.rule.length === 2;
+  }
+  
+  static parseRule(ruleString) {
```

Comment:
hmm is there a pattern to the organization of methods?

maybe we could shift this and the `components` counterpart below right below their constructors since its also somewhat of a constructor

Reply:
I wrote it in the order of usage flow: parsing of the rule, post-parse processing (offset line numbers), test for application, and lastly the application of the rule itself. I feel like this way is intuitive if you want to read the whole class and understand the sequence of process.

Although, the structure check like `isLineSlice` (and `isUnboundedSlice` on the components side) does not fit too well on this scheme. I put it where it was now (on top of everything, after the constructor) if any developers want to understand what are the possible "variants" of the structure possible for this class before reading deep into the class methods.

Not sure if this is the best way, I don't have a strong preference really. What do you think?

Reply:
ok, thanks for the explanation!

How about shifting `isLineRange` to the bottom at least then? Then we can keep the constructors together (parseRule and constructor) while still fitting to your scheme. Likewise for the `Component` version

Reply:
Okay then :+1: